smartsim/_core/launcher/dragon/dragonLauncher.py

```diff
@@ -91,210 +71,47 @@ class DragonLauncher(WLMLauncher):
-                server.send(
-                    DragonBootstrapResponse(dragon_pid=self._dragon_head_process.pid)
-                )
-
-                launcher_socket.close()
-                self._set_timeout(self._timeout)
-                self._handshake(request.address)
-
-                # Only the launcher which started the server is
-                # responsible of it, that's why we register the
-                # cleanup in this code branch.
-                # The cleanup function should not have references
-                # to this object to avoid Garbage Collector lockup
-                server_socket = self._dragon_head_socket
-                server_process_pid = self._dragon_head_process.pid
-
-                # avoid registering the same cleanup more than once
-                if server_socket is not None and self._dragon_head_process is not None:
-                    atexit.register(
-                        _dragon_cleanup,
-                        server_socket=server_socket,
-                        server_process_pid=server_process_pid,
-                        server_authenticator=self._authenticator,
-                    )
-            else:
-                # TODO parse output file
-                log_dragon_outputs()
-                raise LauncherError("Could not receive address of Dragon head process")
+        return self._connector.is_connected
 
     def cleanup(self) -> None:
-        logger.debug("Starting Dragon launcher cleanup")
-        _dragon_cleanup(
-            server_socket=self._dragon_head_socket,
-            server_process_pid=self._dragon_head_pid,
-            server_authenticator=self._authenticator,
-        )
-        self._dragon_head_socket = None
-        self._dragon_head_pid = 0
-        self._authenticator = None
+        self._connector.cleanup()
 
     # RunSettings types supported by this launcher
     @property
     def supported_rs(self) -> t.Dict[t.Type[SettingsBase], t.Type[Step]]:
         # RunSettings types supported by this launcher
-        return {DragonRunSettings: DragonStep, RunSettings: LocalStep}
+        return {
+            DragonRunSettings: DragonStep,
+            SbatchSettings: DragonBatchStep,
+            QsubBatchSettings: DragonBatchStep,
+            RunSettings: LocalStep,
+        }
+
+    def add_step_to_mapping_table(self, name: str, step_map: StepMap) -> None:
+        super().add_step_to_mapping_table(name, step_map)
+
+        if step_map.step_id is None:
+            return
+        sublauncher: t.Optional[t.Union[SlurmLauncher, PBSLauncher]] = None
+        if step_map.step_id.startswith("SLURM-"):
+            sublauncher = self._slurm_launcher
+        elif step_map.step_id.startswith("PBS-"):
+            sublauncher = self._pbs_launcher
+        else:
+            raise ValueError(f"Step id {step_map.step_id} is not valid.")
+
+        sublauncher_step_map = StepMap(
+            step_id=DragonLauncher._unprefix_step_id(step_map.step_id),
```

Comment:
Something feels off about having `def _get_new_id(self) -> str:` live in `dragon_client.py` and `_unprefix_step_id` live in `dragonLauncher.py`. Lets really consider putting them on the step (anywhere they can be colocated) to avoid having to change multiple places if ID formats change...

Reply:
Ditto on the IDs being different (coming from different sources), but feel free to re-raise the question if you are not satisfied with my answer